test/http_downloads_test.rb

```diff
@@ -1,23 +1,41 @@
 require 'test_helper'
+require 'tmpdir'
 
-class Proxy::HttpDownload
-  def command(src, dst)
-    "echo test"
+class HttpDownloadsTest < Test::Unit::TestCase
+  def new_thread_mock
+    thr = mock()
+    thr.stubs(:join).returns(0)
+    thr
   end
-end
 
-class HttpDownloadsTest < Test::Unit::TestCase
-  def test_should_download_a_file
-    assert Proxy::HttpDownloads.start_download 'source', 'destination'
-    assert File.exist?('destination')
+  def tmp(name)
+    File.join(Dir.tmpdir(), "http-download-#{name}.tmp")
+  end
+
+  def test_should_return_an_instance
+    src = tmp('source')
+    dst = tmp('destination')
+    Thread.stubs(:new).returns(new_thread_mock)
+    assert_instance_of Proxy::HttpDownload, Proxy::HttpDownloads.start_download(src, dst)
+  ensure
+    File.delete(dst)
```

Comment:
> Is this necessary? The block we pass to the thread at https://github.com/lzap/smart-proxy/blob/http-download-11318/lib/proxy/util.rb#L15 won't be executed, so no files will be created.

It is, I believe, since our `unlock` code only closes the file handle
(we do create the file to call `flock` on it).

Above comment fixed.

-- 
Later,
 Lukas #lzap Zapletal